### PR TITLE
Revise zoom layout shift fix

### LINKED ISSUE #66297: Resizable Editor: Unintended vertical scrollbar
### Description

In the resizable editor, if there is little content, the vertical scrollbar is not necessary, but it seems to always be displayed.

After investigating this issue, I found that it was caused by the top border added by #66041.

### Step-by-step reproduction instructions

- Open the site editor.
- Open the pattern or template part editor.

### Screenshots, screen recording, code snippet

Image

If I temporarily disable the top border this issue doesn't occur:

Image

### Environment info

_No response_

### Please confirm that you have searched existing issues in the repo.

- [x] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [x] Yes

### PR DESCRIPTION
**Pull request by stokesman**
## What?
A revised way to fix the layout shift that #66041 fixed and an alternative way to fix #66297.

## Why?
The a `border` rule used in the prior PR caused #66297 and also introduces minute layout differences in unintended places. Here’s one @ramonjd spotted https://github.com/WordPress/gutenberg/pull/66041#discussion_r1811577469

https://github.com/user-attachments/assets/f5b2dd18-4ca8-40fc-b6c2-1929e5ebbe79

## How?
Steals Ramon’s idea https://github.com/WordPress/gutenberg/pull/66041#discussion_r1811587699 to use `display: flow-root`

### No layout shift (from #66041)
1. Go to the post editor, zoom out. There should be no layout shift.

### No vertical scrollbar in focused editor
1. Open the site editor.
2. Open the pattern or template part editor.

There should be no vertical scrollbar.

### CODE REVIEW

**Review by andrewserong — COMMENTED** [review]
Overall this is testing great for me, and to me it feels like the _right_ fix. The one issue I've encountered is that this winds up affecting the block previews in global styles, so they're no longer vertically aligned:

| Trunk | This PR |
| --- | --- |
| image | image |
| image | image |

I wonder if it's worth seeing if we can apply `display: flow-root` only to the iframe used for the editor canvas, or something like that?

Otherwise it's testing well:

✅ Resolves the scrollbar issue
✅ Preserves the fix for the layout shift
✅ Working nicely for me in template preview and not template preview modes, in the post and site editors

Thanks again for digging in to try to find a more stable fix!

**Comment by ramonjd** [comment]
Thanks for getting this up 🍺

**Comment by andrewserong** [comment]
> I wonder if it's worth seeing if we can apply display: flow-root only to the iframe used for the editor canvas, or something like that?

Just thinking a little more about this bug. Here's where this CSS rule is overriding the flex rule used in the block previews:

image

https://github.com/WordPress/gutenberg/blob/379e5f42d11a46dfa29fe4c595ba43f1f3ba9b17/packages/block-editor/src/components/inserter/preview-panel.js#L52-L67

So I suppose two options might be:

1. Only apply the `display: flow-root` when we know we want it to be applied (i.e. just on the editor canvas iframe), or
2. Reduce the specificity of the rule so that any other `display` rule will override it

**Comment by stokesman** [comment]
Thanks for the quick review and spotting that issue Andrew! I think to fix it we might incorporate both your ideas. I’d already thought reducing the specificity might be the right thing to do in general. Also, I almost decided to add the style in `VisualEditor` which should keep it applied only for the editor canvas iframe. Here’s the place I was looking at:
https://github.com/WordPress/gutenberg/blob/67fca2ebcc9f0caafccbcaef15021d23cb0bf3dd/packages/editor/src/components/visual-editor/index.js#L355-L366

You’re welcome to jump in and fix it. I can get to it early tomorrow but it’s EOD for me.

**Comment by andrewserong** [comment]
Update: I wonder if it'd work to set this body rule here instead, so it's only on `BlockCanvas` in the visual editor?

https://github.com/WordPress/gutenberg/blob/6d8f1cfba9d1eed0536a0a4b14d035e638051251/packages/editor/src/components/visual-editor/index.js#L348-L359

I see there's also the same rule on the `.is-root-container`, but I assume that's for a slightly different case?

**Comment by andrewserong** [comment]
Snap! We had the same idea 😆 

I'll try pushing a commit and let's see how we go!

**Comment by andrewserong** [comment]
Alrighty, I've given that a try in https://github.com/WordPress/gutenberg/pull/66390/commits/f4eaee5c0c8e56d05022caa55c59d45de401ddc0

It seems to be testing pretty well for me. Feel free to update and/or revert that commit as needed, of course!

**Review by andrewserong — APPROVED** [review]
This is testing well for me, so I'm going to give it the ✅.

It might be worth getting a second opinion before landing since I've worked on this, too! 😅

**Review by ramonjd — APPROVED** [review]
Working well for me. I can't replicate the issues in https://github.com/WordPress/gutenberg/issues/66297 or https://github.com/WordPress/gutenberg/pull/66041

Thank you!

**Comment by andrewserong** [comment]
Thanks for the review! Since we'd like to get this in for 6.7 I'll merge this in. Thanks again for opening this @stokesman 🙇

**Comment by stokesman** [comment]
> I see there's also the same rule on the `.is-root-container`, but I assume that's for a slightly different case?

Yes, I’d been curious about that and found it was added with the Style Book a26d3210e96c2892152429804e06fc062048b476 but there’s nothing there noting what it specifically dealt with and, curiously, doesn’t actually apply in the Style Book now.  It does seem to be the thing saving templated views from potentially having the same layout shift issue as post-only views had. With some brief testing the rule on the body seemed to suffice for that on its own. Something to look at in a separate effort anyway.

Thanks again folks ❤️

**Pull request merged**
